module1/homework/grayscaleImages/test.cpp

```diff
@@ -27,157 +27,87 @@ std::vector<std::pair<uint8_t, uint8_t>> getBitmap(size_t fraction) {
     auto bitmap = getBitmap(fraction);
-
     auto map = decompressGrayscale(bitmap);
     ASSERT_EQ(bitmap.size(), height);
-    for (const auto& row : map) {
-        for (const uint8_t color : row) {
-            EXPECT_EQ(color, 0);
-        }
-    }
+    checkMap(map, fraction);
 }
 
 TEST(compressionTests, ShouldDecompressHalfLines) {
     constexpr size_t fraction = 2;
     auto bitmap = getBitmap(fraction);
-
     auto map = decompressGrayscale(bitmap);
     ASSERT_EQ(map.size(), height);
-    for (const auto& row : map) {
-        for (size_t i = 0; i < row.size() / fraction; ++i) {
-            EXPECT_EQ(row[i], 0);
-        }
-        for (size_t i = row.size() / fraction; i < row.size(); ++i) {
-            EXPECT_EQ(row[i], 1);
-        }
-    }
+    checkMap(map, fraction);
 }
 
 TEST(compressionTests, ShouldDecompressQuaterLines) {
     constexpr size_t fraction = 4;
     auto bitmap = getBitmap(fraction);
-
     auto map = decompressGrayscale(bitmap);
     ASSERT_EQ(map.size(), height);
-    for (const auto& row : map) {
-        for (size_t i = 0; i < row.size() / fraction; ++i) {
-            EXPECT_EQ(row[i], 0);
-        }
-        for (size_t i = row.size() / fraction; i < row.size() / fraction * 2; ++i) {
-            EXPECT_EQ(row[i], 1);
-        }
-        for (size_t i = row.size() / fraction * 2; i < row.size() / (fraction / 3.0); ++i) {
-            EXPECT_EQ(row[i], 2);
-        }
-        for (size_t i = row.size() / (fraction / 3.0); i < row.size(); ++i) {
-            EXPECT_EQ(row[i], 3);
-        }
-    }
+    checkMap(map, fraction);
 }
 
 TEST(compressionTests, ShouldDecompressOneEighthLines) {
     constexpr size_t fraction = 8;
     auto bitmap = getBitmap(fraction);
-
     auto map = decompressGrayscale(bitmap);
     ASSERT_EQ(map.size(), height);
-    for (const auto& row : map) {
-        for (size_t i = 0; i < row.size() / fraction; ++i) {
-            EXPECT_EQ(row[i], 0);
-        }
-        for (size_t i = row.size() / (fraction / 1.0); i < row.size() / (fraction / 2.0); ++i) {
-            EXPECT_EQ(row[i], 1);
-        }
-        for (size_t i = row.size() / (fraction / 2.0); i < row.size() / (fraction / 3.0); ++i) {
-            EXPECT_EQ(row[i], 2);
-        }
-        for (size_t i = row.size() / (fraction / 3.0); i < row.size() / (fraction / 4.0); ++i) {
-            EXPECT_EQ(row[i], 3);
-        }
-        for (size_t i = row.size() / (fraction / 4.0); i < row.size() / (fraction / 5.0); ++i) {
-            EXPECT_EQ(row[i], 4);
-        }
-        for (size_t i = row.size() / (fraction / 5.0); i < row.size() / (fraction / 6.0); ++i) {
-            EXPECT_EQ(row[i], 5);
-        }
-        for (size_t i = row.size() / (fraction / 6.0); i < row.size() / (fraction / 7.0); ++i) {
-            EXPECT_EQ(row[i], 6);
-        }
-        for (size_t i = row.size() / (fraction / 7.0); i < row.size(); ++i) {
-            EXPECT_EQ(row[i], 7);
-        }
-    }
+    checkMap(map, fraction);
```

Comment:
nice. Now when you have only fraction which could change. You can try to parameterize test.